### ISSUE TITLE
训练时has_contrastive_loss参数相关的定义问题

### ISSUE THREAD

**Issue by wsssh3** [XavierJiezou/D2LS#12]
作者您好！我在使用LoveDA数据集训练时，遇到了AttributeError: 'tuple' object has no attribute 'log_softmax'的报错，查看问题原因时发现是下面代码：
    def training_step(self, batch, batch_idx):
        img, mask = batch['img'], batch['gt_semantic_seg']
        loss = 0
        prediction = self.net(img)
        if self.config.get("has_contrastive_loss",False):
            loss += prediction[-1]
            prediction = prediction[:-1]
        loss += self.loss(prediction, mask)
生成的prediction是一个三元组，包含dynamic_output, basci_output, contrastive_loss。但是这里的has_contrastive_loss定义为false，config文件中也没有相关定义，我想知道这一部分为什么这样设置？这并不能正确的训练，还请指教

**Comment by caixiaoshun** [comment]
你好，这里network/models/d2ls.py文件中DynamicDictionaryLearning类has_contrastive_loss参数默认值已经更改为True,请拉取最新的代码

**Comment by wsssh3** [comment]
感谢！您好，我想问一下grass和cloud数据集的test是如何做的呢？论文中放出的相关结果是val数据集的指标吗？但是在定性可视化时给出的图是验证集还是测试集呢？期待您的回复

<img width="837" height="648" alt="Image" src="https://github.com/user-attachments/assets/f80aec9c-7c87-4c71-92e4-2bdfb72626ca" />

**Comment by caixiaoshun** [comment]
Please refer to #11 regarding the test set split.
